File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    findKthLargest(nums, k) {
        const maxHeap = new MaxPriorityQueue();
        for (const num of nums) {
            maxHeap.enqueue(Number(num));
        }
        
        let res = 0;
        for (let i = 0; i < k; i++) {
            res = maxHeap.dequeue();
        }

        return res;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-2.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        res = nums[0]
        curr_sum = 0

        for n in nums:
            curr_sum = max(curr_sum, 0)
            curr_sum += n
            res = max(res, curr_sum)

        return res
        